### PR TITLE
- Added Spring Security to project;

### PR DESCRIPTION
**Pull request by MykolaFatkullin**
- Changed type fo date in all of DTOs;
- Deleted redundant fields and classes

### CODE REVIEW

File: src/main/java/com/tickets/annotation/EmailValidation.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.hibernate.validator.internal.constraintvalidators.hv.EmailValidator;
+
+@Documented
+@Constraint(validatedBy = EmailValidator.class)
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface EmailValidation {
+    String message() default "Invalid phone number";
```
Comment (mfedenko09) [inline src/main/java/com/tickets/annotation/EmailValidation.java]:
```suggestion
    String message() default "Invalid email";
```

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/tickets/annotation/EmailValidation.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/com/tickets/annotation/EmailValidation.java]: agree

File: src/main/java/com/tickets/model/dto/CinemaHallRequestDto.java [hunk 2]
```diff
@@ -1,7 +1,12 @@
 package com.tickets.model.dto;
 
+import com.sun.istack.NotNull;
```
Comment (boroda4436) [inline src/main/java/com/tickets/model/dto/CinemaHallRequestDto.java]: import javax.validation.constraints.NotNull;

Comment (boroda4436) [reply src/main/java/com/tickets/model/dto/CinemaHallRequestDto.java]: Have you tested your code?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/tickets/model/dto/CinemaHallRequestDto.java (hunk 2 above)
Comment (MykolaFatkullin) [reply src/main/java/com/tickets/model/dto/CinemaHallRequestDto.java]: @boroda4436 yes. Maybe, I didn't test it correctly

**Review by boroda4436 — APPROVED** [review]

File: src/main/java/com/tickets/service/authentication/impl/AuthenticationObjectProcessingImpl.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.tickets.service.authentication.impl;
+
+import com.tickets.service.authentication.AuthenticationObjectProcessing;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthenticationObjectProcessingImpl implements AuthenticationObjectProcessing {
```
Comment (boroda4436) [inline src/main/java/com/tickets/service/authentication/impl/AuthenticationObjectProcessingImpl.java]: you can just call `authentication.getName()` and do not implement this class

**Pull request merged**